app/scripts/modules/kubernetes/src/v2/pipelines/stages/patchManifest/patchManifestConfig.controller.ts

```diff
@@ -0,0 +1,78 @@
+import { IController, IScope } from 'angular';
+import { IPatchOptions, MergeStrategy } from './patchOptionsForm.component';
+import {
+  IKubernetesManifestCommandData,
+  IKubernetesManifestCommandMetadata,
+  KubernetesManifestCommandBuilder,
+} from '../../../manifest/manifestCommandBuilder.service';
+import { ExpectedArtifactService, IExpectedArtifact } from '@spinnaker/core';
+import { loadAll } from 'js-yaml';
+
+export class KubernetesV2PatchManifestConfigCtrl implements IController {
+  public state = {
+    loaded: false,
+  };
+
+  public metadata: IKubernetesManifestCommandMetadata;
+  public textSource = 'text';
+  public artifactSource = 'artifact';
+  public sources = [this.textSource, this.artifactSource];
+
+  public expectedArtifacts: IExpectedArtifact[];
+
+  constructor(private $scope: IScope) {
+    'ngInject';
+
+    this.expectedArtifacts = ExpectedArtifactService.getExpectedArtifactsAvailableToStage(
+      $scope.stage,
+      $scope.$parent.pipeline,
+    );
+
+    const defaultOptions: IPatchOptions = {
+      mergeStrategy: MergeStrategy.strategic,
+      record: true,
+    };
+
+    if (this.$scope.stage.isNew) {
+      this.$scope.stage.options = defaultOptions;
+    }
+
+    KubernetesManifestCommandBuilder.buildNewManifestCommand(
+      this.$scope.application,
+      this.$scope.stage.patchBody,
+      this.$scope.stage.moniker,
+    ).then((builtCommand: IKubernetesManifestCommandData) => {
```

Comment:
The compiler should infer `IKubernetesManifestCommandData` here - no need to explicitly type `builtCommand`.